xonsh/base_shell.py

```diff
@@ -159,6 +159,9 @@ def default(self, line):
         src, code = self.push(line)
         if code is None:
             return
+
+        events.on_precommand.fire(src)
+
```

Comment:
PEP8 says to use these sparingly, and I agree because you can get more code on the screen simeltaneously the fewer blank lines you have.  I don't think this blocks merging.